Code/autopkglib/SparkleUpdateInfoProvider.py

```diff
@@ -118,27 +125,88 @@ class SparkleUpdateInfoProvider(Processor):
         },
     }
 
+    def prepare_curl_cmd(self, url, headers=None):
+        """Assemble curl command and return it."""
+
+        curl_cmd = [
+            super(SparkleUpdateInfoProvider, self).curl_binary(),
+            "--location",
+            "--compressed",
+        ]
+        super(SparkleUpdateInfoProvider, self).add_curl_common_opts(curl_cmd)
+        if headers:
```

Comment:
This logic is already handled in URLGetter.py's `add_curl_common_opts` - why do it here again?

Reply:
Similiar reply as in https://github.com/autopkg/autopkg/pull/548#discussion_r319856830

These headers are not from Processor `self.env.get("request_headers")`. `SparkleUpdateInfoProvider` does not have `request_headers` input variable. Instead there is a `appcast_request_headers` variable which is specific to `SparkleUpdateInfoProvider` only.

We could change the name of the variable but that would break many recipes https://github.com/search?q=appcast_request_headers&type=Code.